### PR TITLE
restore a subset of the rand module

### LINKED ISSUE #706: Security Implications of #675 (Delete rand.py)
In #675, the rand module was removed from the codebase, and users of the module were instructed to use os.urandom instead. However, using os.urandom is not an alternative for the whole rand module, just for some functions like rand.bytes.

There are other functions in the module, such as rand.add, that are changing the PRNG state of OpenSSL. Using such functions is of particular importance for fork safety, as described here: https://wiki.openssl.org/index.php/Random_fork-safety

and here:
https://docs.python.org/2/library/ssl.html#multi-processing

Please consider reverting #675, or at least bringing back the minimum subset of the rand module that will allow pyopenssl users to deal with fork safety.

### PR DESCRIPTION
**Pull request by reaperhulk**
fixes #706 

We should bikeshed on the changelog language probably.

### CODE REVIEW

**Comment by alex** [comment]
The error stuff appers unused?

File: src/OpenSSL/rand.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
+"""
+PRNG management routines, thin wrappers.
+"""
+
+from OpenSSL._util import lib as _lib
+
+
+_builtin_bytes = bytes
```
Comment (alex) [inline src/OpenSSL/rand.py]: This isn't needed, just use `bytes` below.

File: src/OpenSSL/rand.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+def status():
+    """
+    Check whether the PRNG has been seeded with enough data.
+
+    :return: 1 if the PRNG is seeded enough, 0 otherwise.
+    """
+    return _lib.RAND_status()
+
+
+def cleanup():
```
Comment (alex) [inline src/OpenSSL/rand.py]: Do we really need this?

File: tests/test_rand.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+    def test_status(self):
+        """
+        `OpenSSL.rand.status` returns `1` if the PRNG has sufficient entropy,
+        `0` otherwise.
+        """
+        # It's hard to know what it is actually going to return.  Different
+        # OpenSSL random engines decide differently whether they have enough
+        # entropy or not.
+        assert rand.status() in (0, 1)
```
Comment (alex) [inline tests/test_rand.py]: _Surely_ this should actually always be `1`?

File: src/OpenSSL/rand.py (hunk 2 above)
Comment (reaperhulk) [reply src/OpenSSL/rand.py]: Probably not. As of 1.1.0 this is already a noop

File: tests/test_rand.py (hunk 3 above)
Comment (reaperhulk) [reply tests/test_rand.py]: Maybe it's possible to trigger a 0 when calling cleanup and not adding enough back...? I don't know. This is just a restoration of the code that was previously there, but this test is not particularly useful. I'd be fine changing it to assert 1. Plus as an added bonus we'll get to find out some insane edge case if it turns out 1 isn't always true in the tests.
